client/src/components/Search/SearchBox.js

```diff
@@ -12,131 +12,136 @@ import TagPropType from '../Tag/TagPropType';
+  onToggleFilter,
+  onHideFilter,
+  onChange,
+  onHide,
+  onTagDelete,
+  onTagClick,
+  placeholder = i18n._t('Admin.SEARCH', 'Search'),
+  expanded,
+  formId,
+  id,
+  searchText,
+  hideable,
+  showFilters,
+  name,
+  dirty,
+  clearable,
+  tagData = [],
+  borders = {
+    top: false,
+    right: false,
+    bottom: true,
+    left: true,
+  },
+  children
+}) => {
+  const [hasFocus, setHasFocus] = useState(false);
+  const [width, setWidthState] = useState(window.innerWidth - 180 - 55);
+  const [tagWidth, setTagWidth] = useState(0);
+  const nodeRef = useRef(null);
+  const isFirstRender = useRef(true);
+
+  const getComponentWidth = () => {
+    if (!nodeRef.current) {
+      return 0;
+    }
+    return nodeRef.current.getBoundingClientRect().width;
+  };
 
-  componentDidUpdate() {
-    const width = this.getComponentWidth();
-    this.setWidth(width);
-  }
+  const setWidth = (newWidth) => {
+    setWidthState((prevWidth) => {
+      if (prevWidth !== newWidth) {
+        return newWidth;
+      }
+      return prevWidth;
+    });
+  };
 
   /**
    * Detect resizing of the component.
    * If it's bigger, the summary view is triggered.
    * @param tagListDimension
    */
-  onResize(dimension) {
-    this.setWidth(dimension.width);
-  }
-
-  onTagListResize(dimensions) {
-    const tagWidth = dimensions.width;
-    if (this.state.tagWidth !== tagWidth) {
-      this.setState({ tagWidth });
-    }
-  }
-
-  setWidth(width) {
-    if (this.state.width !== width) {
-      this.setState({ width });
-    }
-  }
-
-  /**
-   * Measure the width of the component.
-   * @returns number
-   */
-  getComponentWidth() {
-    const node = this.nodeRef;
-    if (!node) {
-      return 0;
-    }
-
-    return node.getBoundingClientRect().width;
-  }
+  const onResize = (dimension) => {
+    setWidth(dimension.width);
+  };
+
+  const onTagListResize = (dimensions) => {
+    const newTagWidth = dimensions.width;
+    setTagWidth((prevTagWidth) => {
+      if (prevTagWidth !== newTagWidth) {
+        return newTagWidth;
+      }
+      return prevTagWidth;
+    });
+  };
 
   /**
    * Calculate our input's left padding base on the width of our tag list.
    * @returns {number}
    */
-  calculateInputLeftPadding() {
+  const calculateInputLeftPadding = () => {
     // Account for if the search icon is hidden in compact mode
-    const existingPadding = this.state.width > 576 ? 55 : 20;
-    return this.state.tagWidth + existingPadding;
-  }
+    const existingPadding = width > 576 ? 55 : 20;
+    return tagWidth + existingPadding;
+  };
 
-  calculateInputRightPadding() {
-    // Right padding is used to set the alignedment of the "x" that shows in chrome
-    // when you type in the search box
-    return this.state.width < 576 ? 121 : 264;
-  }
+  // Right padding is used to set the alignedment of the "x" that shows in chrome
+  // when you type in the search box
+  const calculateInputRightPadding = () => (width < 576 ? 121 : 264);
```

Comment:
If you're moving the comment to above the function please use `/**  */` style comments